mentat/code_file.py

```diff
@@ -0,0 +1,48 @@
+import math
+from pathlib import Path
+
+
+class Interval:
+    def __init__(
+        self,
+        start: int,
+        end: int,
+    ):
+        self.start = start
+        self.end = end
+
+    def contains(self, k):
+        return self.start <= k <= self.end
+
+
+def parse_intervals(interval_string: str) -> list[Interval] | None:
+    try:
+        intervals = []
+        for interval in interval_string.split(","):
+            interval = interval.split("-", 1)
+            if len(interval) == 1:
+                intervals += [Interval(int(interval[0]), int(interval[0]))]
+            else:
+                intervals += [Interval(int(interval[0]), int(interval[1]))]
+        return intervals
+    except (ValueError, IndexError):
+        return None
+
+
+class CodeFile:
+    def __init__(self, path: str | Path):
```

Comment:
let's add a docstring here so it's people reading the codebase to understand what this class does